doc/designs/adtrust/sidconfig.md

```diff
@@ -0,0 +1,287 @@
+
+
+## Design
+
+Installers and SID-related options:
+- the options `--add-sids`, `--netbios-name`, `--rid-base` and 
+`--secondary-rid-base` are moved from ADTrustInstallInterface to a separate
+new InstallInterface: SIDInstallInterface.
+- ADTrustInstallInterface now inherits from SIDInstallInterface.
+- `adtrustinstance.ADTRUSTInstance.__init__` now accepts an additional
+parameter: `fulltrust`. When set to True, the class ADTRUSTInstance configures
+the trust as usual. When set to False, only the SID-related part is executed.
+- `ipa-server-install` and `ipa-replica-install` now always call
+`adtrust.install_check` and `adtrust.install`, but the method is using the
+provided options (especially `options.setup_adtrust`) to know if full
+configuration is required or only the SID-related part.
+
+The `ipa-adtrust-install` code is already written in order to be
+idempotent (can be called multiple times, even with different values), and
+this property is of great value for this feature. It allows to keep
+the changes as minimal as possible to the existing installers, and
+call `ipa-adtrust-install` even if the SID-part is already setup.
+
+New command to enable SID generation after an upgrade:
+as we don't want to automatically enable SID generation during an upgrade,
+a new command is provided in order to manually enable the feature. The command
+only requires an admin ticket (no need to be root) and relies on the admin
+framework.
+
+The command uses a mechanism similar to server_conncheck:
+- the user must have Replication Administrators privilege
+- the user launches an ipa command communicating with IPA framework
+- the admin framework uses Dbus to get access a DBus interface that
+launches a command through oddjob, allowing the command to run as root.
+The oddjob command is a wrapper around `adtrustinstance.ADTRUSTInstance`.
+
+## Implementation
+
+- Dependencies: no additional dependency on ipa-server package
+- Backup and Restore: no new file to backup
+
+## Feature Management
+
+### UI
+
+No new UI for server/replica installation.
+
+`IPA server / ID ranges` tab is already available and displays Primary and 
+Secondary RID base.
+
+Open questions:
+- should we display the SID identifier / NetBIOS Name in
+`IPA Server / Configuration` tab instead of `IPA Server / Trusts / Global Trust`
+Configuration tab?
+- how can we show that SID generation is enabled / not enabled in the WebUI?
+
+### CLI
+
+| Command |	Options |
+| --- | ----- |
+| ipa-server-install | --netbios-name=NETBIOS_NAME --rid-base=RID_BASE --secondary-rid-base=SECONDARY_RID_BASE  |
+| ipa-replica-install | --netbios-name=NETBIOS_NAME --rid-base=RID_BASE --secondary-rid-base=SECONDARY_RID_BASE --add-sids |
+| ipa config-mod | --enable-sid --add-sids |
+
+The `--enable-sid` option turns the feature on, and `--add-sids` triggers
+the SIDgen task in order to immediately generate SID for existing users or
+groups.
+`--add-sids` can be used only when `--enable-sid` is used.
```

Comment:
I think what you meant here is that `--add-sids` option depends on the presence of `--enable-sid` but `--enable-sid` can be used without it. Is that correct?

Perhaps, we can move detection logic to the framework and simply allow to specify either option. If SIDs weren't enabled or `--enable-sid` was not specified in the same request, `--add-sids` would simply fail. If `--enable-sid` was  specified, we enable SID generation. The key is to be able to detect whether SID generation is enabled.

While `ipa idrange-mod` is used to add RID bases, we still need a way to specify NetBIOS name for the domain. Should this be part of `ipa config-mod --enable-sid`? I think this part needs more investigation.